salt/python_apps.sls

```diff
@@ -50,6 +53,9 @@ include:
     - cwd: {{ directory }}
     - require:
       - virtualenv: {{ directory }}/.ve
+      {% if grains.osmajorrelease|string == '22' %}
+      - pkg: libffi-dev
+      {% endif %}
```

Comment:
This isn't required by all our Python apps, and cove.sls already includes python.extensions before python_apps.

In extensions.sls, we can roll up the libffi-dev state into the "python c extensions" state.

```suggestion
```

Reply:
I saw errors from `pip` without this explicit requirement.
However, I can't replicate this issue now.
I've added this suggestion in the latest commit.